apps/cli/src/store/local/actions.ts

```diff
@@ -333,31 +349,72 @@ export async function processProject(
     ),
   );
   // Handle all cross references
-  await Promise.all(pages.map((page) => postProcessMdast(session, { file: page.file })));
-  // Write all pages
   await Promise.all(
     pages.map((page) =>
-      writeFile(session, {
+      postProcessMdast(session, {
         file: page.file,
-        projectSlug: siteProject.slug,
-        pageSlug: page.slug,
+        strict: opts?.strict,
+        checkLinks: opts?.checkLinks,
       }),
     ),
   );
+  // Write all pages
+  if (writeFiles) {
+    await Promise.all(
+      pages.map((page) =>
+        writeFile(session, {
+          file: page.file,
+          projectSlug: siteProject.slug,
+          pageSlug: page.slug,
+        }),
+      ),
+    );
+  }
   log.info(toc(`📚 Built ${pages.length} pages for ${siteProject.slug} in %s.`));
+  return project;
 }
 
 export async function processSite(session: ISession, opts?: ProcessOptions): Promise<boolean> {
   if (opts?.reloadConfigs) loadAllConfigs(session);
   const siteConfig = selectors.selectLocalSiteConfig(session.store.getState());
   session.log.debug(`Site Config:\n\n${yaml.dump(siteConfig)}`);
   if (!siteConfig?.projects.length) return false;
-  await Promise.all(
+  const projects = await Promise.all(
     siteConfig.projects.map((siteProject) => processProject(session, siteProject, opts)),
   );
-  await writeSiteManifest(session);
-  // Copy all assets
-  copyLogo(session, siteConfig.logo);
-  siteConfig.actions.forEach((action) => copyActionResource(session, action));
+  if (opts?.strict) {
+    const hasWarnings = projects
```

Comment:
Hah, the details of this block are super hard to parse - but I think that doesn't matter at all. It's just printing all warnings and errors, then throwing - I guess only if there are errors? Warnings are ok even for strict?